Upgrade to angular 19 migration guide + 1022 breaking changes list (#3261)
SoftwareAG/c8y-docs: content/web/upgrade-bundle/upgrading-to-angular-19.md

> ---

Author: adding unresolved conversation not to forget about this -> https://github.com/Cumulocity-IoT/c8y-docs/pull/3261#issuecomment-2829521849

Author: 1022 is released, so this PR can be merged when approved.

title: Upgrading from Angular 18 to Angular 19  
layout: redirect
weight: 430
---

Angular 19 is supported from version `1022.0.0`. The following configuration changes are required before you can run the application:

{{< c8y-admon-important >}}
Since Angular 19, each component decorator must have `standalone` property with value `true` or `false`. It also applies
to module federation plugins.
{{< /c8y-admon-important >}}

> - Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 18.

Reviewer: ```suggestion
- Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 19.
```

---
title: Upgrading from Angular 18 to Angular 19
layout: redirect
weight: 430
---

> Angular 19 is supported from version `1022.0.0`. The following configuration changes are required before you can run the application:

Reviewer 2: Version of what?

Author: added 'of WEB Sdk'

> Angular 19 is supported from version `1022.0.0`. The following configuration changes are required before you can run the application:

Author: ```suggestion
Angular 19 is supported from version `1022.0.0` of Web SDK. The following configuration changes are required before you can run the application:
```

> Angular 19 is supported from version `1022.0.0` of Web SDK. The following configuration changes are required before you can run the application:

Reviewer 2: ```suggestion
Starting with version 1022.0.0, the Web SDK supports Angular 19. The following configuration changes are required before you can run the application:
```

- Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 19.
{{< c8y-admon-important >}}
Angular directives, components and pipes are now standalone by default. Specify `standalone: false` for declarations that are currently declared in an NgModule. 

> It also applies to module federation plugins, including plugins that uses earlier versions of Angular, but are utilized in application migrated to Angular 19.

Reviewer 2: ```suggestion
This also applies to module federation plugins, including plugins that use earlier versions of Angular, but are utilized in applications migrated to Angular 19.
```

{{< /c8y-admon-important >}}
- Update all `@c8y` dependencies to version `1022.x.x` in your *package.json*.
- Update `ngx-bootstrap` to version `19.0.2`.
- Update `@angular/cdk` to version `19.x.x`.
- `Node.js`, `TypeScript`, `RxJS`: [Version compatibility](https://angular.dev/reference/versions#actively-supported-versions).
- Follow the `Angular 19` upgrade guide: [Updating to version 19](https://angular.dev/update-guide?v=18.0-19.0&l=2).

> The Web SDK version 1022.0.0 introduces multiple breaking changes:

Reviewer 2: Only one breaking change is listed?

Author: no, there will be more, team is informed that they should add their breaking changes. this PR will be merged right after release, but it's happening in some weeks from now

Author: switching to "Draft" so it's more clear that it's not ready to be merged.

> With version 1022.0.0 there will be multiple breaking changes introduced:

Reviewer 2: ```suggestion
The Web SDK version 1022.0.0 introduces multiple breaking changes:
```

> - The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable which leads to necessity of adding `rootContext: ViewContext.Dashboard` to context dashboard routes. See more at [Support for cross application dashboard import/export functionality](/change-logs/#ui-c8y-1022-0-0-dashboard-cross-application-copy-paste)

Reviewer 2: ```suggestion
- The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable. This requires adding `rootContext: ViewContext.Dashboard` to the context dashboard routes. For details, refer to [Support for cross application dashboard import/export functionality](/change-logs/#ui-c8y-1022-0-0-dashboard-cross-application-copy-paste).
```

layout: redirect
weight: 430
---

Starting with version 1022.0.0, the Web SDK supports Angular 19. The following configuration changes are required before you can run the application:

- Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 19.
{{< c8y-admon-important >}}
Angular directives, components and pipes are now standalone by default. Specify `standalone: false` for declarations that are currently declared in an NgModule. 

This also applies to module federation plugins, including plugins that use earlier versions of Angular, but are utilized in applications migrated to Angular 19.
{{< /c8y-admon-important >}}
- Update all `@c8y` dependencies to version `1022.x.x` in your *package.json*.
- Update `ngx-bootstrap` to version `19.0.2`.
- Update `@angular/cdk` to version `19.x.x`.
- `Node.js`, `TypeScript`, `RxJS`: [Version compatibility](https://angular.dev/reference/versions#actively-supported-versions).
- Follow the `Angular 19` upgrade guide: [Updating to version 19](https://angular.dev/update-guide?v=18.0-19.0&l=2).

The Web SDK version 1022.0.0 introduces multiple breaking changes:

> - The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable. This requires adding `rootContext: ViewContext.Dashboard` to the context dashboard routes. For details, refer to [Support for cross application dashboard import/export functionality](/change-logs/#ui-c8y-1022-0-0-dashboard-cross-application-copy-paste).

Reviewer 2: We should not link to change log entries for details. It should rather be the other way round, details should be included in the documentation. This applies to this item and the next one.

Reviewer: Usually the announcement in the changelog is what is created before the documentation change is done.. Should we then have a similar text in both locations or remove the details from the change log announcement and place it in the documentation?

Author: @BeateRixen please answer to Tristan's question as we want to merge this PR as soon as possible (1022 was released om 16 june, so it's already one week ago)

Reviewer 2: @reey If we do not have the documentation available yet when creating the change log entry, we should provide a similar text in both locations. But we should not have details in the change log only. Of course, it is better to have the documentation available with the change log entry :-).

Author: done for both items. the one related to tabs was refactored a little bit to emphasize hooking tabs generic functionality (not the Import/export functionality)

Reviewer 2: @jdre-c8y The indentation under breaking changes does not look correct (bulleted list). Either indent all content under each item or use headings instead of list items.

Author: @BeateRixen true, it was not looking good. I decided to go with headings. how about now?

Reviewer 2: Looks much better! I added one more heading and approved.

layout: redirect
weight: 430
---

Starting with version 1022.0.0, the Web SDK supports Angular 19. The following configuration changes are required before you can run the application:

- Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 19.
{{< c8y-admon-important >}}
Angular directives, components and pipes are now standalone by default. Specify `standalone: false` for declarations that are currently declared in an NgModule. 

This also applies to module federation plugins, including plugins that use earlier versions of Angular, but are utilized in applications migrated to Angular 19.
{{< /c8y-admon-important >}}
- Update all `@c8y` dependencies to version `1022.x.x` in your *package.json*.
- Update `ngx-bootstrap` to version `19.0.2`.
- Update `@angular/cdk` to version `19.x.x`.
- `Node.js`, `TypeScript`, `RxJS`: [Version compatibility](https://angular.dev/reference/versions#actively-supported-versions).
- Follow the `Angular 19` upgrade guide: [Updating to version 19](https://angular.dev/update-guide?v=18.0-19.0&l=2).

The Web SDK version 1022.0.0 introduces multiple breaking changes:

> - The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable. This requires adding `rootContext: ViewContext.Dashboard` to the context dashboard routes. 

Reviewer 2: @jdre-c8y  Shouldn´t it be "has been refactored" here?

Author: of course, fixed now.

> - The dashboard setting component will be refactored to use a secondary router outlet in order to make these type of views hookable. This requires adding `rootContext: ViewContext.Dashboard` to the context dashboard routes. 

Author: ```suggestion
- The dashboard setting component has been refactored to use a secondary router outlet in order to make these type of views hookable. This requires adding `rootContext: ViewContext.Dashboard` to the context dashboard routes. 
```

> The reason for it is that new **Import/Export** tab will be added to the dashboard settings with `hookTab` function. This generic solution allows to add tabs to named tabs outlets. 

Reviewer 2: ```suggestion
The reason for this is that the new **Import/Export** tab will be added to the dashboard settings with the `hookTab` function. This generic solution allows adding tabs to named tab outlets. 
```

> The reason for this is that the new **Import/Export** tab will be added to the dashboard settings with the `hookTab` function. This generic solution allows adding tabs to named tab outlets. 

Author: ```suggestion
The reason for this is that the new **Import/Export** tab has been added to the dashboard settings with the `hookTab` function. This generic solution allows adding tabs to named tab outlets. 
```

This specific tab allows to export dashboards to JSON files, import dashboards from previously exported JSON files and edit the dashboard in an editor. It allows to copy dashboards across applications.

> Example of how tab and route for this tab can be added:

Reviewer 2: ```suggestion
Example of how the tab and route for this tab can be added:
```

---
title: Upgrading from Angular 18 to Angular 19
layout: redirect
weight: 430
---

Starting with version 1022.0.0, the Web SDK supports Angular 19. The following configuration changes are required before you can run the application:

- Run the command `ng update @angular/core@19 @angular/cli@19` to update Angular core and CLI to version 19.
{{< c8y-admon-important >}}
Angular directives, components and pipes are now standalone by default. Specify `standalone: false` for declarations that are currently declared in an NgModule. 

This also applies to module federation plugins, including plugins that use earlier versions of Angular, but are utilized in applications migrated to Angular 19.
{{< /c8y-admon-important >}}
- Update all `@c8y` dependencies to version `1022.x.x` in your *package.json*.
- Update `ngx-bootstrap` to version `19.0.2`.
- Update `@angular/cdk` to version `19.x.x`.
- `Node.js`, `TypeScript`, `RxJS`: [Version compatibility](https://angular.dev/reference/versions#actively-supported-versions).
- Follow the `Angular 19` upgrade guide: [Updating to version 19](https://angular.dev/update-guide?v=18.0-19.0&l=2).

> The Web SDK version 1022.0.0 introduces multiple breaking changes:

Reviewer 2: ```suggestion
### Breaking changes
The Web SDK version 1022.0.0 introduces multiple breaking changes.
```
